klippy/extras/filament_motion_sensor.py

```diff
@@ -36,12 +40,31 @@ def __init__(self, config):
                 self._handle_not_printing)
         self.printer.register_event_handler('idle_timeout:idle',
                 self._handle_not_printing)
+        # Register GCODE commands
+        self.gcode = self.printer.lookup_object('gcode')
+        self.gcode.register_command(
+            'FILAMENT_MOTION_SENSOR', self.cmd_FILAMENT_MOTION_SENSOR,
+            desc=self.cmd_FILAMENT_MOTION_SENSOR_help)
+    cmd_FILAMENT_MOTION_SENSOR_help = "Enable debug logging for the encoder event"
+    def cmd_FILAMENT_MOTION_SENSOR(self, gcmd):
+        options = collections.OrderedDict({
```

Comment:
Also are you sure that using dictionary to enable verbose/debug is necessary?

Reply:
It's not the same thing multiply the detection_lenght.
I currently use the BTT Smart Sensor V2, that not need to be calibrated, the pin goes up every 3mm forward o backward(retraction).
I suppose to check if the extruder try to push forward the filament and that can't go out for a clog.
On the first check i can determine that the filament was pushed for more than 3mm and nothing happen, no trigger from the sensor. This can happen because there's a partial clog, the extruder does not have enough force to resolve the clog and loose steps, the retraction does not trigger the sensor or the past calculated position can't be precise.
 
On the next "time" checks, i need to know if the distance is again greater than 3mm not 6,9,12 mm.

I think the calibration of this kind of sensor is useful if you don't know the correct detection_lenght to play with it. 
The real work can be the accuracy of the past position, but it's too difficult and not so useful; or the consider the trigger when the filament goes forward that is different when the trigger goes backward for a retraction.

Reply:
For the verbose/debug it's not important. I have used it to understand what happen in print, how that difference change and understand how to perform better, if it perform better! :P